src/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -64,11 +67,14 @@ See the LICENSE file in the project root for more information.
 
     <SharedLibrary Condition="'$(OS)' == 'Windows_NT'">$(FrameworkLibPath)\Framework$(LibFileExt)</SharedLibrary>
     <SharedLibrary Condition="'$(OS)' != 'Windows_NT'">$(FrameworkLibPath)\libframework$(LibFileExt)</SharedLibrary>
+    <DynamicBuildPropertyDependencies Condition="'$(CalledViaPackage)' == 'true'">SetupProperties</DynamicBuildPropertyDependencies>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(IlcCompileDependsOn)'=='' and '$(NativeCompilationDuringPublish)' != 'false'">
     <IlcCompileDependsOn Condition="'$(BuildingFrameworkLibrary)' != 'true'">Compile;ComputeIlcCompileInputs</IlcCompileDependsOn>
     <IlcCompileDependsOn Condition="'$(IlcMultiModule)' == 'true' and '$(BuildingFrameworkLibrary)' != 'true'">$(IlcCompileDependsOn);BuildFrameworkLib</IlcCompileDependsOn>
+    <IlcCompileDependsOn Condition="'$(TargetOS)' == 'Windows_NT'">$(IlcCompileDependsOn);SetupWindowsProps</IlcCompileDependsOn>  
+    <IlcCompileDependsOn Condition="'$(TargetOS)' != 'Windows_NT'">$(IlcCompileDependsOn);SetupUnixProps</IlcCompileDependsOn>  
```

Comment:
Would it make sense for SetupWindowsProps/SetupUnixProps to have same name on both Windows and Unix?

Reply:
I'm not sure. Ultimately only one set of  `.props` will be loaded at any given build, so it would make sense to abstract that detail away from `Native.targets` and avoid another OS check. I chose specifically-named build targets to make build logs explicit - property setup would be easily identifiable.

Reply:
It is not unusual to have the same target defined in multiple files. For example, CoreCompile is a well known target that works like that.